JMH benchmarks in swirlds-benchmarks are broken

It may be related to recent build changes (or maybe not, but not very likely).

Steps to reproduce:

* Run a clean JMH build in `platform-sdk` using `../gradlew jmhJar`
* In `platform-sdk/swirlds-benchmarks`, run JMH using `java -cp build/libs/swirlds-benchmarks-0.42.0-SNAPSHOT-jmh.jar org.openjdk.jmh.Main CryptoBenchMerkleDb.transferSerial`

Output:

```
java.lang.ExceptionInInitializerError
	at com.swirlds.config.api.ConfigurationBuilder.create(ConfigurationBuilder.java:139)
	at com.swirlds.benchmark.BaseBench.loadConfig(BaseBench.java:90)
	at com.swirlds.benchmark.BaseBench.setup(BaseBench.java:110)
	at com.swirlds.benchmark.jmh_generated.CryptoBenchMerkleDb_transferSerial_jmhTest._jmh_tryInit_f_cryptobenchmerkledb0_0(CryptoBenchMerkleDb_transferSerial_jmhTest.java:379)
	at com.swirlds.benchmark.jmh_generated.CryptoBenchMerkleDb_transferSerial_jmhTest.transferSerial_AverageTime(CryptoBenchMerkleDb_transferSerial_jmhTest.java:145)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:475)
	at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:458)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: No ConfigurationBuilderFactory implementation found!
	at com.swirlds.config.api.intern.ConfigurationProvider.loadImplementation(ConfigurationProvider.java:47)
	at com.swirlds.config.api.intern.ConfigurationProvider.<init>(ConfigurationProvider.java:36)
	at com.swirlds.config.api.intern.ConfigurationProvider.<clinit>(ConfigurationProvider.java:31)
	... 17 more
```

I found that the JMH jar file, `swirlds-benchmarks-0.42.0-SNAPSHOT-jmh.jar` doesn't contain an entry for `com.swirlds.config.api.spi.ConfigurationBuilderFactory` in its META-INF/services. This is probably why ServiceLoader can't find any implementations in runtime (`ConfigurationProvider` above uses ServiceLoader mechanism).

--- a/platform-sdk/swirlds-benchmarks/build.gradle.kts
+++ b/platform-sdk/swirlds-benchmarks/build.gradle.kts
@@ -28,6 +28,7 @@ jmhModuleInfo {
     requires("com.swirlds.virtualmap")
     requires("jmh.core")
     requires("org.apache.logging.log4j")
+    runtimeOnly("com.swirlds.config.impl")
 }
 
 jmh {

